diff --git a/configure.ac b/configure.ac
index cc69015b10..c77e92affc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -873,11 +873,7 @@ fi
 rm -f conftest.c conftest.out
 
 AC_MSG_CHECKING([for multiarch])
-AS_CASE([$ac_sys_system],
-  [Darwin*], [MULTIARCH=""],
-  [FreeBSD*], [MULTIARCH=""],
-  [MULTIARCH=$($CC --print-multiarch 2>/dev/null)]
-)
+MULTIARCH=
 AC_SUBST([MULTIARCH])
 AC_MSG_RESULT([$MULTIARCH])
 
