diff --git a/configure.ac b/configure.ac
index ba768aea93..031d5c897f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7288,13 +7288,7 @@ dnl sets MODULE_$NAME_CFLAGS and MODULE_$NAME_LDFLAGS
 AC_DEFUN([PY_STDLIB_MOD], [
   AC_MSG_CHECKING([for stdlib extension module $1])
   m4_pushdef([modcond], [MODULE_]m4_toupper([$1]))dnl
-  m4_pushdef([modstate], [py_cv_module_$1])dnl
-  dnl Check if module has been disabled by PY_STDLIB_MOD_SET_NA()
-  AS_IF([test "$modstate" != "n/a"], [
-    AS_IF([m4_ifblank([$2], [true], [$2])],
-       [AS_IF([m4_ifblank([$3], [true], [$3])], [modstate=yes], [modstate=missing])],
-       [modstate=disabled])
-  ])
+  m4_pushdef([modstate], [disabled])dnl
   _MODULE_BLOCK_ADD(modcond[_STATE], [$modstate])
   AS_VAR_IF([modstate], [yes], [
     m4_ifblank([$4], [], [_MODULE_BLOCK_ADD([MODULE_]m4_toupper([$1])[_CFLAGS], [$4])])
@@ -7312,9 +7306,7 @@ dnl PY_STDLIB_MOD_SIMPLE([NAME], [CFLAGS], [LDFLAGS])
 dnl cflags and ldflags are optional
 AC_DEFUN([PY_STDLIB_MOD_SIMPLE], [
   m4_pushdef([modcond], [MODULE_]m4_toupper([$1]))dnl
-  m4_pushdef([modstate], [py_cv_module_$1])dnl
-  dnl Check if module has been disabled by PY_STDLIB_MOD_SET_NA()
-  AS_IF([test "$modstate" != "n/a"], [modstate=yes])
+  m4_pushdef([modstate], [disabled])dnl
   AM_CONDITIONAL(modcond, [test "$modstate" = yes])
   _MODULE_BLOCK_ADD(modcond[_STATE], [$modstate])
   AS_VAR_IF([modstate], [yes], [
