diff --git a/Makefile.pre.in b/Makefile.pre.in
index 09ceccda1d..91e08cf28f 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -2769,6 +2769,12 @@ update-config:
 
 Python/thread.o: @THREADHEADERS@ $(srcdir)/Python/condvar.h
 
+write-python-for-build:
+	echo "#!/bin/sh" > python-for-build
+	echo "set -e" >> python-for-build
+	echo "exec env $(PYTHON_FOR_BUILD) \$$@" >> python-for-build
+	chmod +x python-for-build
+
 ##########################################################################
 # Module dependencies and platform-specific files
 
