diff --git a/Modules/posixmodule.c b/Modules/posixmodule.c
index 12f72f525f..4503c5fc60 100644
--- a/Modules/posixmodule.c
+++ b/Modules/posixmodule.c
@@ -326,6 +326,13 @@ corresponding Unix manual entries for more information on calls.");
 #  endif  /* _MSC_VER */
 #endif  /* ! __WATCOMC__ || __QNX__ */

+#if __APPLE__
+#include <TargetConditionals.h>
+#if TARGET_OS_IPHONE
+#    undef HAVE_SYSTEM
+#endif
+#endif
+
 _Py_IDENTIFIER(__fspath__);

 /*[clinic input]
