diff --git a/configure.ac b/configure.ac
index c62a565eb6..5a5a076f06 100644
--- a/configure.ac
+++ b/configure.ac
@@ -178,24 +178,13 @@ AC_MSG_CHECKING([for Python interpreter freezing])
 AC_MSG_RESULT([$PYTHON_FOR_FREEZE])
 AC_SUBST([PYTHON_FOR_FREEZE])
 
-AS_VAR_IF([cross_compiling], [yes],
-  [
-    dnl external build Python, freezing depends on Programs/_freeze_module.py
-    FREEZE_MODULE_BOOTSTRAP='$(PYTHON_FOR_FREEZE) $(srcdir)/Programs/_freeze_module.py'
-    FREEZE_MODULE_BOOTSTRAP_DEPS='$(srcdir)/Programs/_freeze_module.py'
-    FREEZE_MODULE='$(FREEZE_MODULE_BOOTSTRAP)'
-    FREEZE_MODULE_DEPS='$(FREEZE_MODULE_BOOTSTRAP_DEPS)'
-    PYTHON_FOR_BUILD_DEPS=''
-  ],
-  [
-    dnl internal build tools also depend on Programs/_freeze_module and _bootstrap_python.
-    FREEZE_MODULE_BOOTSTRAP='./Programs/_freeze_module'
-    FREEZE_MODULE_BOOTSTRAP_DEPS="Programs/_freeze_module"
-    FREEZE_MODULE='$(PYTHON_FOR_FREEZE) $(srcdir)/Programs/_freeze_module.py'
-    FREEZE_MODULE_DEPS="_bootstrap_python \$(srcdir)/Programs/_freeze_module.py"
-    PYTHON_FOR_BUILD_DEPS='$(BUILDPYTHON)'
-  ]
-)
+dnl external build Python, freezing depends on Programs/_freeze_module.py
+FREEZE_MODULE_BOOTSTRAP='$(PYTHON_FOR_FREEZE) $(srcdir)/Programs/_freeze_module.py'
+FREEZE_MODULE_BOOTSTRAP_DEPS='$(srcdir)/Programs/_freeze_module.py'
+FREEZE_MODULE='$(FREEZE_MODULE_BOOTSTRAP)'
+FREEZE_MODULE_DEPS='$(FREEZE_MODULE_BOOTSTRAP_DEPS)'
+PYTHON_FOR_BUILD_DEPS=''
+
 AC_SUBST([FREEZE_MODULE_BOOTSTRAP])
 AC_SUBST([FREEZE_MODULE_BOOTSTRAP_DEPS])
 AC_SUBST([FREEZE_MODULE])
